discordbot/OverworldDiscordBot.py

```diff
@@ -0,0 +1,282 @@
+from functools import lru_cache
+from threading import Thread
+from time import sleep
+from enum import Enum
+from distutils.util import strtobool
+
+import discord
+import pokebase as pb
+from discord.ext import commands
+from discord.utils import get
+from PIL import Image
+
+sys.path.append('../')
+from lookups import Util
+from nxreader import SWSHReader
+# from . import Channels
+
+class Channels(Enum):
+    NoChannel = 0
+    EventNotificationChannelIdForShiny = 1
+    EventNotificationChannelIdForBrilliant = 2
+    EventNotificationChannelIdForMark = 3
+    EventNotificationChannelIdForRareMark = 4
+    NotificationChannelForInfo = 5
+    TextNotificationChannelIdForShiny = 6
+    TextNotificationChannelIdForBrilliant = 7
+    TextNotificationChannelIdForMark = 8
+    TextNotificationChannelIdForRareMark = 9
+
+class OverworldDiscordBot(commands.Bot):
+    def __init__(self):
+        # config to refactor out later
+        # prefix for bot commands
+        bot_prefix = "$"
+        # set up commands.Bot with specified prefix
+        super().__init__(command_prefix=bot_prefix)
+
+        # default values for variables as to not throw errors
+        self.thread = None
+        self.thread_running = True
+        self.reader = None
+        self.config = None
+        self.is_configured = False
+
+        # function to run when the bot is "ready"
+        @self.event
+        async def on_ready():
+            # tell user in console that the bot is ready
+            _message = "Overworld Discord Bot has started."
+            await self.send_discord_msg(_message, Channels.NotificationChannelForInfo)
+            # change bot presence to "Watching some ram for shinies"
+            await self.change_presence(activity=discord.Activity(type=discord.ActivityType.watching, name="some ram for shinies"))
+
+        # function to run whenever the start command is called
+        @self.command()
+        async def start(ctx):
+            if not self.is_configured:
+                _message = "Unable to start bot due to missing configuration! Exiting."
+                await self.send_discord_msg(_message, Channels.NotificationChannelForInfo)
+                return
+            # create reader object
+            try:
+                self.reader = SWSHReader(self.config["IP"])
+                # create and start a thread for reading from the switch
+                self.thread = Thread(target=self.reader_func,args=(ctx,))
+                self.thread.start()
+            except socket.timeout:
+                _message = "Unable to connect to IP {}. Check that the Switch is available.".format(self.config["IP"])
+                await self.send_discord_msg(_message, Channels.NotificationChannelForInfo)
+
+        # function to run whenever the ping command is called
+        @self.command()
+        async def ping(ctx):
+            # say the bots latency in discord channel
+            await ctx.send(f'Pong! {round(self.latency, 3)}')
```

Comment:
Note for future: use MS instead of self.latency

Reply:
Can we do this here and now? What would this look like?

Reply:
this can be done by either doing something like seen [here](https://github.com/poketwo/poketwo/blob/8b0ba184dfde87298d63602cd23982491a7f4737/cogs/bot.py#L339), or it can be done by converting the number given  by [.latency](https://discordpy.readthedocs.io/en/stable/api.html?highlight=latency#discord.Client.latency) to milliseconds

Reply:
I'm going to skip this for now. If you want to do this go ahead!